src/spaceObjects/spaceship.cpp

```diff
@@ -706,6 +707,17 @@ void SpaceShip::collide(Collisionable* other, float force)
             docking_offset = sf::rotateVector(getPosition() - other->getPosition(), -other->getRotation());
             float length = sf::length(docking_offset);
             docking_offset = docking_offset / length * (length + 2.0f);
+            // if Player has no permission to dock give a penalty of -10 reputation
+            P<SpaceStation> station = dock_object;
+            if (station && !station->hasDockingPermission(this)) {
+                P<PlayerSpaceship> player = P<SpaceShip>(this);
+                player->removeReputationPoints(10);
```

Comment:
need to check `if(player)`
it might be some bad AI trying to dock a cpu ship, then the game will crash